Templates/templates/PHP/Model/ComplexType.php.tt

```diff
@@ -131,12 +131,26 @@ foreach(var property in complex.Properties.Where(prop => prop.Type.GetTypeString
     * <#=property.GetSanitizedLongDescription()#>
 <# } #>
     *
-    * @return <#=propertyTypeReference#>|null The <#=propertyName#>
+    * @return <#=propertyTypeReference#><# if (property.IsCollection()){ #>[]<# } #>|null The <#=propertyName#>
     */
     public function get<#=propertyName.ToCheckedCase()#>()
     {
-        if (array_key_exists("<#=property.Name.ToCamelize()#>", $this->_propDict)) {
-            if (is_a($this->_propDict["<#=camelCasePropertyName#>"], "<#=fullPropertyTypeName#>") || is_null($this->_propDict["<#=camelCasePropertyName#>"])) {
+        if (array_key_exists("<#=property.Name.ToCamelize()#>", $this->_propDict) && !is_null($this->_propDict["<#=camelCasePropertyName#>"])) {
+    <# if (property.IsCollection()) { #>   
+            if (count($this->_propDict['<#=camelCasePropertyName#>']) === 0) {
+              return $this->_propDict['<#=camelCasePropertyName#>'];
+            }
+            if (is_a($this->_propDict['<#=camelCasePropertyName#>'][0], ' <#=propertyTypeReference#>')) {
```

Comment:
There is a space that needs to be removed here after the first single quote: `' <#=propertyTypeReference#>'`

L143 checks if the first entry is the generated type, and if so, it returns the array. Under which circumstances would we  the foreach loop happen?

Reply:
Noted will remove the space.

The foreach loop happens when the property list isn't of that given type which might occur if we just pass an array of associative arrays instead of an array of that given type. The issue with these type checks is that we are doing them in the wrong place(that's what has been there) since we don't enforce type constraints when setting the values in the setter.